pilot/proxy/envoy/config.go

```diff
@@ -906,6 +910,77 @@ func buildEgressHTTPRoutes(mesh *proxyconfig.MeshConfig, node proxy.Node,
 	return httpConfigs.normalize()
 }
 
+// buildEgressTCPListeners builds a listener on 0.0.0.0 per each distinct port of all TCP egress
+// rules and a cluster per each TCP egress rule
+func buildEgressTCPListeners(mesh *proxyconfig.MeshConfig, node proxy.Node,
+	config model.IstioConfigStore) (Listeners, Clusters) {
+
+	tcpListeners := make(Listeners, 0)
+	tcpClusters := make(Clusters, 0)
+
+	if node.Type == proxy.Router {
+		// No egress rule support for Routers. As semantics are not clear.
+		return tcpListeners, tcpClusters
+	}
+
+	egressRules, errs := model.RejectConflictingEgressRules(config.EgressRules())
+
+	if errs != nil {
+		glog.Warningf("Rejected rules: %v", errs)
+	}
+
+	tcpRulesByPort := make(map[int][]*proxyconfig.EgressRule)
+
+	for _, rule := range egressRules {
+		for _, port := range rule.Ports {
+			protocol := model.Protocol(strings.ToUpper(port.Protocol))
+			if protocol != model.ProtocolTCP {
```

Comment:
as I noted earlier, we need to allow tcp, redis, mongo, and https as well.

Reply:
@rshriram see my previous reply. I prefer to work PR by PR, first support for TCP, then support for mongodb and redis.